src/Compiler/Driver/CompilerDiagnostics.fs

```diff
@@ -2073,7 +2074,7 @@ let FormatDiagnosticLocation (tcConfig: TcConfig) m : FormattedDiagnosticLocatio
             | DiagnosticStyle.Default ->
                 let file = file.Replace('/', Path.DirectorySeparatorChar)
                 let m = withStart (mkPos m.StartLine (m.StartColumn + 1)) m
-                (sprintf "%s(%d,%d): " file m.StartLine m.StartColumn), m, file
+                (sprintf "◦→ %s (%d,%d)" file m.StartLine m.StartColumn), m, file
```

Comment:
Some context - one of the reason the compiler emits this `<file>(<range>): <message>` format is that this is the [MSBuild Canonical Diagnostic format](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-diagnostic-format-for-tasks?view=vs-2022) - emitting this format helps many tools that operate in MSBuild or use MSBuild (like VS/VSCode/Rider) seamlessly be able to report diagnostics without any additional integration work.

Reply:
Yeah I think we need a more `Rich` diagnostic format the current one is not really friendly and I think we can do better.